src/Mobs/Monster.h

```diff
@@ -74,6 +76,30 @@ class cMonster :
 	virtual void CheckEventSeePlayer(cChunk & a_Chunk);
 	virtual void EventSeePlayer(cPlayer * a_Player, cChunk & a_Chunk);
 
+	/** Returns whether the mob can be leashed. */
+	bool CanBeLeashed() const { return m_CanBeLeashed; }
+
+	/** Sets whether the mob ban be leashed, for extensibility in plugins */
+	void SetCanBeLeashed(bool a_CanBeLeashed) { m_CanBeLeashed = a_CanBeLeashed; }
+
+	/** Returns whether the monster is leashed to an entity. */
+	bool IsLeashed() const { return (m_LeashedTo != nullptr); }
+
+	/** Leash the monster to a entity. */
+	void SetLeashedTo(cEntity * a_Entity);
+
+	/** Unleash the monster from a entity. */
+	void SetUnleashed(bool a_DropPickup);
+
+	/** Returns the entity to where is leashed this mob, returns nullptr if it's not leashed */
+	cEntity * GetLeashedTo() const { return m_LeashedTo; }
+
+	/** Sets entity position to where is leashed this mob */
+	void SetLeashToPos(Vector3d * pos) { m_LeashToPos = pos; }
+
+	/** Gets entity position to where mob should be leashed */
+	Vector3d * GetLeashToPos() const { return m_LeashToPos; }
+
```

Comment:
It would make sense exporting some of these to Lua API. Use the `// tolua_begin` and `// tolua_end` markers to auto-export them.